test/evp_extra_test2.c

```diff
@@ -450,6 +456,56 @@ static int test_dh_paramfromdata(void)
 
 #endif
 
+static int test_new_keytype(void)
```

Comment:
Suggestion: we don't seem to be doing it in this file for some reason, but generally I like to include a comment prior to the test function declaration briefly outlining the main purpose of the test, e.g. in this case "Test that calling EVP_PKEY_Q_keygen() for a non-standard keytype works as expected", or similar.

Reply:
added.